app/models/deploy.rb

```diff
@@ -4,7 +4,8 @@ class Deploy < ActiveRecord::Base
   belongs_to :stage, touch: true
   belongs_to :build
   belongs_to :job
-  belongs_to :buddy, class_name: 'User'
+  belongs_to :buddy, -> { unscope(where: "deleted_at") }, class_name: 'User'
+  belongs_to :stage_with_deleted, -> { unscope(where: "deleted_at") }, class_name: 'Stage', foreign_key: 'stage_id'
```

Comment:
uh nice ... so it removes the scope on that column ... funky

On Wed, Jul 27, 2016 at 7:16 PM, Robert Ikeoka <notifications@github.com>
wrote:

> In app/models/deploy.rb
> <https://github.com/zendesk/samson/pull/1131#discussion_r72556016>:
>
> > @@ -4,7 +4,8 @@ class Deploy < ActiveRecord::Base
> >    belongs_to :stage, touch: true
> >    belongs_to :build
> >    belongs_to :job
> > -  belongs_to :buddy, class_name: 'User'
> > +  belongs_to :buddy, -> { unscope(where: "deleted_at") }, class_name: 'User'
> > +  belongs_to :stage_with_deleted, -> { unscope(where: "deleted_at") }, class_name: 'Stage', foreign_key: 'stage_id'
>
> It removes the "WHERE stages.deleted_at IS NULL" from the query. So it
> would get both non-deleted and deleted stages. If the desired is deleted
> stage, it would need unscope(where: "deleted_at").where("stages.deleted_at
> IS NOT NULL") instead.
>
> I didn't want to update the regular stages scope, since that would cause
> some wierd UI bugs, but changing the buddy association was fine, since
> generally if people are looking back in time, they would still want to know
> who the buddy is even though the buddy is no longer an active user in
> Samson.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1131/files/188714baf38f6a4a9cadec2f58016756e4f9262d#r72556016>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ61G4rL4prRBzzZM-ScJSFovnOjUks5qaBD7gaJpZM4JIewY>
> .
>